app/grandchallenge/components/views.py

```diff
@@ -337,3 +342,75 @@ def get_queryset(self):
         return queryset.prefetch_related(
             "values", "values__image", "values__interface"
         )
+
+
+class CIVSetBulkDeleteView(LoginRequiredMixin, FormView):
+    form_class = CIVSetDeleteForm
+    model = None
+    template_name = "components/civ_set_delete_confirm.html"
+
+    @property
+    def base_object(self):
+        raise NotImplementedError
+
+    def get_queryset(self, *args, **kwargs):
+        # subset by permission
+        permission = f"{self.model._meta.app_label}.delete_{self.model._meta.model_name}"
+        return get_objects_for_user(
+            user=self.request.user,
+            perms=[permission],
+            klass=self.base_object.civ_sets_related_manager.all(),
+        )
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update(
+            {
+                "base_object": self.base_object,
+                "delete_count": (
+                    self.selected_objects.count()
+                    if self.selected_objects
+                    else 0
+                ),
+            }
+        )
+        return context
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update(
+            {
+                "queryset": (
+                    self.selected_objects
+                    if self.selected_objects
+                    else self.get_queryset()
+                ),
+            }
+        )
+        return kwargs
+
+    @cached_property
+    def selected_objects(self):
+        # Selecting happens on the ListView through a GET request to this view
+        # on POST the selected objects are part of the form field
+        if self.request.method == "GET":
+            delete_all = self.request.GET.get("delete-all", None)
+            if delete_all:
+                return self.get_queryset()
+            else:
+                selected = self.request.GET.getlist(
+                    "selected-for-deletion", None
+                )
+                # filtering the original queryset
+                # so that only objects with permission are included
+                return self.get_queryset().filter(pk__in=selected)
+        else:
+            return None

```

Comment:
You could instead return `self.model.objects.none()` so that you're always dealing with a queryset downstream, that would probably remove the need for `self.selected_objects.count() if self.selected_objects else 0`.

Reply:
Ah good idea!